src/mlpack/methods/ann/convolution_rules/svd_convolution.hpp

```diff
@@ -77,23 +79,24 @@ class SVDConvolution
       // advantage of the low rank.
       if (rank * (filter.n_rows + filter.n_cols) < filter.n_elem)
       {
-        arma::Mat<eT> subFilter = V.unsafe_col(0) * s(0);
+        MatType subFilter = V.unsafe_col(0) * s(0);
         NaiveConvolution<BorderMode>::Convolution(input, subFilter, subOutput);
 
         subOutput = subOutput.t();
-        NaiveConvolution<BorderMode>::Convolution(subOutput, U.unsafe_col(0),
+        MatType tmp = arma::conv_to<MatType>::from(U.unsafe_col(0));
+        NaiveConvolution<BorderMode>::Convolution(subOutput, tmp,
```

Comment:
This is necessary since it was passing a `Col` and now only accepts MatType, which should always be a `MatType`. I do not think this should have some performance issue, hopefully not.

if you have a better proposal, please let me know

Reply:
It doesn't work to just leave `U.unsafe_col(0)` here?  That should return `MatType` (for Armadillo types).

There will later need to be a rethink here, since Bandicoot doesn't implement `unsafe_col()` and I'm not sure it makes sense for it to.